src/coniql/app.py

```diff
@@ -1,49 +1,75 @@
-import traceback
 from argparse import ArgumentParser
 from pathlib import Path
-from typing import Any, Dict
+from typing import List, Optional
 
 import aiohttp_cors
+import strawberry
 from aiohttp import web
-from tartiflette import Engine, TartifletteError
-from tartiflette_aiohttp import register_graphql_handlers
+from graphql import GraphQLError
+from strawberry.aiohttp.views import GraphQLView
+from strawberry.subscriptions import GRAPHQL_TRANSPORT_WS_PROTOCOL, GRAPHQL_WS_PROTOCOL
+from strawberry.types import ExecutionContext
 
-from coniql.caplugin import CAPlugin
-from coniql.plugin import PluginStore
-from coniql.pvaplugin import PVAPlugin
-from coniql.simplugin import SimPlugin
+import coniql.strawberry_schema as schema
 
 from . import __version__
 
 
-async def error_coercer(exception: Exception, error: Dict[str, Any]) -> Dict[str, Any]:
-    if isinstance(exception, TartifletteError):
-        e = exception.original_error
+class ConiqlSchema(strawberry.Schema):
+    # Override to remove stack trace printer with every log
+    def process_errors(
+        self,
+        errors: List[GraphQLError],
+        execution_context: Optional[ExecutionContext] = None,
+    ) -> None:
+        for error in errors:
+            print(error.message)
+
+
+def create_schema(debug: bool):
+    # Create the schema
+    if debug:
+        return strawberry.Schema(
+            query=schema.Query,
+            subscription=schema.Subscription,
+            mutation=schema.Mutation,
+        )
     else:
-        e = exception
-    if e:
-        traceback.print_exception(type(e), e, e.__traceback__)
-    return error
+        return ConiqlSchema(
+            query=schema.Query,
+            subscription=schema.Subscription,
+            mutation=schema.Mutation,
+        )
+
 
+def create_app(use_cors: bool, debug: bool):
+    # Create the schema
+    strawberry_schema = create_schema(debug)
 
-def make_engine() -> Engine:
-    engine = Engine(
-        sdl=Path(__file__).resolve().parent / "schema.gql",
-        error_coercer=error_coercer,
-        modules=["coniql.resolvers"],
+    # Create the GraphQL view to attach to the app
+    view = GraphQLView(
+        schema=strawberry_schema,
+        subscription_protocols=[GRAPHQL_TRANSPORT_WS_PROTOCOL, GRAPHQL_WS_PROTOCOL],
```

Comment:
I agree that:
- It should be off for production
- It should be possible to enable it for testing

Reply:
If you run locally, try going to `http://localhost:8080/ws`. This is the end point that we configure in the application and is where you can access the GraphIQL interface.

Reply:
Thanks, I can access it from that URL.

Reply:
I've added a command line argument `--graphiql` which will enable the GraphiQL interface, but by default (and so for production) it will be disabled. Commit https://github.com/dls-controls/coniql/pull/44/commits/eaa0f347e991116dbd1508b30dc97d5fa6114784.